BUILDING.md

```diff
@@ -0,0 +1,21 @@
+Building
+--------
+
+You can choose to build this library either with `cmake` or `automake`.
+The pre-requisite for compilations are:
+
+- you should have cloned this repository with `git clone https://github.com/libsigcplusplus/libsigcplusplus.git`
+- you should have a modern version of `cmake` or `automake` depending on your preference
```

Comment:
This is too vague. I would just leave this and the GCC or Clang mention out. The build script will complain if it doesn't have what it needs.